core/src/impl/Kokkos_Abort.cpp

```diff
@@ -18,6 +18,9 @@
 #define KOKKOS_IMPL_PUBLIC_INCLUDE
 #endif
 
+// include cstddef to deal with internal system header issue on some builds
+// where our CI complained about missing definition of std::size_t
+#include <cstddef>
```

Comment:
We should try to workaround the issue elsewhere.

Reply:
https://github.com/kokkos/kokkos/pull/8455 should fix the issue

Reply:
@masterleinad changes reverted with [489ae29](https://github.com/kokkos/kokkos/pull/8447/commits/489ae29adb6b86e5aac03f0c2a59ee0f213f9c37) 👍